### ISSUE TITLE
Not able to resolve interface of `LifetimeScope.Single` in child scope

### ISSUE THREAD

**Issue by quabug** [hadashiA/VContainer#179]
version: 1.7.0
unity: 2020.3.1f1

``` c#
using UnityEngine;
using VContainer;

public interface IFoo {}
public class Foo : IFoo {}

public interface IBar {}
public class Bar : IBar
{
    public Bar(IFoo _) {}
}

public class TestScope : MonoBehaviour
{
    void Awake()
    {
        var builder = new ContainerBuilder();
        var container = builder.Build();
        container.CreateScope(b =>
        {
            b.Register<Foo>(Lifetime.Singleton).AsImplementedInterfaces();
            b.Register<Bar>(Lifetime.Singleton).AsImplementedInterfaces();
            b.RegisterBuildCallback(c => c.Resolve<IBar>()); // error: VContainerException: Failed to resolve Bar : No such registration of type: IFoo
        });
    }
}
```

```
VContainerException: Failed to resolve Bar : No such registration of type: IFoo
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Internal/ReflectionInjector.cs:46)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Internal/Registration.cs:44)
VContainer.Container+<>c__DisplayClass5_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:152)
System.Lazy`1[T].CreateValue () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.Lazy`1[T].LazyInitValue () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.Lazy`1[T].get_Value () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
VContainer.Container.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:174)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:69)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:72)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.Resolve[T] (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/IObjectResolverExtensions.cs:8)
TestScope+<>c.<Awake>b__0_1 (VContainer.IObjectResolver c) (at Assets/TestScope.cs:23)
VContainer.ContainerBuilder.EmitCallbacks (VContainer.IObjectResolver container, System.Collections.Generic.IEnumerable`1[T] callbacks) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/ContainerBuilder.cs:158)
VContainer.ScopedContainerBuilder.BuildScope () (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/ContainerBuilder.cs:39)
VContainer.ScopedContainer.CreateScope (System.Action`1[T] installation) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:88)
VContainer.Container.CreateScope (System.Action`1[T] installation) (at Library/PackageCache/jp.hadashikick.vcontainer@1.7.0/Runtime/Container.cs:189)
TestScope.Awake () (at Assets/TestScope.cs:19)
```

**Comment by quabug** [comment]
A quick debug show's the problem possible be here:
The existence check shouldn't just check the `ImplementType` but the existence of registration itself.
![image](https://user-images.githubusercontent.com/683655/115102378-09dae180-9f7d-11eb-8513-d3d9783ce380.png)

**Comment by hadashiA** [comment]
That's a bad bug..
Thanks for the report and the source code suggestion.

 Fixed in #181 and released [v1.7.1](https://github.com/hadashiA/VContainer/releases/tag/1.7.1)
